src/lib/components/TokenSelect/index.tsx

```diff
@@ -45,10 +46,25 @@ interface TokenSelectDialogProps {
 }
 
 export function TokenSelectDialog({ value, onSelect }: TokenSelectDialogProps) {
+  const { chainId } = useActiveWeb3React()
+  const [previousChainId, setPreviousChainId] = useState(chainId)
+
   const [query, setQuery] = useState('')
   const queriedTokens = useQueryCurrencies(query)
   const tokens = useMemo(() => queriedTokens?.filter((token) => token !== value), [queriedTokens, value])
```

Comment:
You can't base `listHasMoreTokensThanJustNativeAsset` on `tokens` because it's already filtered by the query, and no tokens for a query has a different UI ("No results found"). I think you'll need to update useQueryCurrencies to take a token list, and then get the token list here to check if there are tokens on the network.

Also not that native asset is not technically a token, so you could feasibly call it something like `listHasTokens`.